jerry-core/lit/lit-strings.cpp

```diff
@@ -828,31 +989,34 @@ lit_compare_utf8_strings (const lit_utf8_byte_t *string1_p, /**< utf-8 string */
  * @return true - if first string is less than second string,
  *         false - otherwise.
  */
-bool lit_compare_utf8_strings_relational (const lit_utf8_byte_t *string1_p, /**< utf-8 string */
+bool lit_compare_cesu8_strings_relational (const lit_utf8_byte_t *string1_p, /**< utf-8 string */
                                           lit_utf8_size_t string1_size, /**< string size */
                                           const lit_utf8_byte_t *string2_p, /**< utf-8 string */
                                           lit_utf8_size_t string2_size) /**< string size */
 {
-  lit_utf8_iterator_t iter1 = lit_utf8_iterator_create (string1_p, string1_size);
-  lit_utf8_iterator_t iter2 = lit_utf8_iterator_create (string2_p, string2_size);
+  lit_utf8_byte_t *string1_pos = (lit_utf8_byte_t *) string1_p;
```

Comment:
In some cases it could be useful, but it would also over complicate others. In lots of cases we know   the string is only ASCII, or we expect it to be ASCII. If we don't use iterators, then we can easily take advantage of these situations.